framework/src/parser/MooseSyntax.C

```diff
@@ -62,12 +63,16 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   /// Variable/AuxVariable Actions
   syntax.registerActionSyntax("AddVariableAction", "Variables/*");
+  syntax.registerSyntaxType("Variables/*", "VariableName");
+  syntax.registerSyntaxType("Variables/*", "NonlinearVariableName");
```

Comment:
Right, but you can't just replace `NonlinearVariableName` with `VariableName` either. You need to support multiple associations.